src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/ExtractOriginalAlignmentRecordsByNameSpark.java

```diff
@@ -82,14 +87,21 @@ protected void runTool( final JavaSparkContext ctx ) {
 
         final Broadcast<Set<String>> namesToLookForBroadcast = ctx.broadcast(parseReadNames());
 
-        final JavaRDD<GATKRead> reads =
-                getUnfilteredReads().filter(read -> namesToLookForBroadcast.getValue().contains(read.getName())).cache();
+        final Function<GATKRead, Boolean> predicate = getGatkReadBooleanFunction(namesToLookForBroadcast);
```

Comment:
Super-picky, ignore if you like: "predicate" is a confusing name to me in this context, presumably meaning "to assert" in this context. Since it's an object, I might suggest "nameFilter" which indicates what it is. Again, this is just a suggestion, feel free to ignore.

Reply:
This is a predicate (noun form) meaning a boolean valued function of one argument. https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html

Reply:
Ah, I did some googling but missed that definition. I now see it's super-standard, so I retract this comment.

Reply:
Thanks for clarifying, Chris!
One note to add: returning a `Function` makes it serializable, which we can use in Spark.
I'm being lazy :see_no_evil: by NOT implementing a class that implements both `Predicate` and `Serializable`